awscli/customizations/overridesslcommonname.py

```diff
@@ -0,0 +1,93 @@
+# Copyright 2022 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License"). You
+# may not use this file except in compliance with the License. A copy of
+# the License is located at
+#
+#     http://aws.amazon.com/apache2.0/
+#
+# or in the "license" file accompanying this file. This file is
+# distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF
+# ANY KIND, either express or implied. See the License for the specific
+# language governing permissions and limitations under the License.
+
+import os
+
+DEFAULT_DNS_SUFFIX = 'amazonaws.com'
+CHINA_DNS_SUFFIX = 'amazonaws.com.cn'
+GOV_DNS_SUFFIX = 'sc2s.sgov.gov'
+
+common_regions = [
```

Comment:
Where did this list come from?

Reply:
i had to manually put it together by combing through the `endpoints.json` file. Obviously far from ideal and mistake prone, but I'm not sure there's a better way. I suppose I could put a script together that runs through every region/service combo and compile a list of ones that use SSL common name. I can work on that to validate.

Reply:
ok so I compiled a list programmatically (can share code if desired) to come up with a full list of services and regions that have an SSL common name. This doesn't account for ones that do have the correct format `{service}.{region}.{dnsSuffix}` which I tried to exclude.  Do we want to just override them all? List below with `service` being the key and `(region, endpoint_url)` being the value:

```txt
rds: (us-east-1, rds.amazonaws.com)
elasticmapreduce: (af-south-1, af-south-1.elasticmapreduce.amazonaws.com)
elasticmapreduce: (ap-east-1, ap-east-1.elasticmapreduce.amazonaws.com)
elasticmapreduce: (ap-northeast-1, ap-northeast-1.elasticmapreduce.amazonaws.com)
elasticmapreduce: (ap-northeast-2, ap-northeast-2.elasticmapreduce.amazonaws.com)
elasticmapreduce: (ap-northeast-3, ap-northeast-3.elasticmapreduce.amazonaws.com)
elasticmapreduce: (ap-south-1, ap-south-1.elasticmapreduce.amazonaws.com)
elasticmapreduce: (ap-southeast-1, ap-southeast-1.elasticmapreduce.amazonaws.com)
elasticmapreduce: (ap-southeast-2, ap-southeast-2.elasticmapreduce.amazonaws.com)
elasticmapreduce: (ap-southeast-3, ap-southeast-3.elasticmapreduce.amazonaws.com)
elasticmapreduce: (ca-central-1, ca-central-1.elasticmapreduce.amazonaws.com)
elasticmapreduce: (eu-central-1, elasticmapreduce.eu-central-1.amazonaws.com)
elasticmapreduce: (eu-north-1, eu-north-1.elasticmapreduce.amazonaws.com)
elasticmapreduce: (eu-south-1, eu-south-1.elasticmapreduce.amazonaws.com)
elasticmapreduce: (eu-west-1, eu-west-1.elasticmapreduce.amazonaws.com)
elasticmapreduce: (eu-west-2, eu-west-2.elasticmapreduce.amazonaws.com)
elasticmapreduce: (eu-west-3, eu-west-3.elasticmapreduce.amazonaws.com)
elasticmapreduce: (me-south-1, me-south-1.elasticmapreduce.amazonaws.com)
elasticmapreduce: (sa-east-1, sa-east-1.elasticmapreduce.amazonaws.com)
elasticmapreduce: (us-east-1, elasticmapreduce.us-east-1.amazonaws.com)
elasticmapreduce: (us-east-2, us-east-2.elasticmapreduce.amazonaws.com)
elasticmapreduce: (us-west-1, us-west-1.elasticmapreduce.amazonaws.com)
elasticmapreduce: (us-west-2, us-west-2.elasticmapreduce.amazonaws.com)
rds: (us-east-1, rds.amazonaws.com)
rds: (us-east-1, rds.amazonaws.com)
shield: (aws-global, shield.us-east-1.amazonaws.com)
shield: (aws-global, shield.us-east-1.amazonaws.com)
shield: (aws-global, shield.us-east-1.amazonaws.com)
shield: (aws-global, shield.us-east-1.amazonaws.com)
shield: (aws-global, shield.us-east-1.amazonaws.com)
shield: (aws-global, shield.us-east-1.amazonaws.com)
shield: (aws-global, shield.us-east-1.amazonaws.com)
shield: (aws-global, shield.us-east-1.amazonaws.com)
shield: (aws-global, shield.us-east-1.amazonaws.com)
shield: (aws-global, shield.us-east-1.amazonaws.com)
shield: (aws-global, shield.us-east-1.amazonaws.com)
shield: (aws-global, shield.us-east-1.amazonaws.com)
shield: (aws-global, shield.us-east-1.amazonaws.com)
shield: (aws-global, shield.us-east-1.amazonaws.com)
shield: (aws-global, shield.us-east-1.amazonaws.com)
shield: (aws-global, shield.us-east-1.amazonaws.com)
shield: (aws-global, shield.us-east-1.amazonaws.com)
shield: (aws-global, shield.us-east-1.amazonaws.com)
shield: (aws-global, shield.us-east-1.amazonaws.com)
shield: (aws-global, shield.us-east-1.amazonaws.com)
shield: (aws-global, shield.us-east-1.amazonaws.com)
shield: (aws-global, shield.us-east-1.amazonaws.com)
sqs: (af-south-1, af-south-1.queue.amazonaws.com)
sqs: (ap-east-1, ap-east-1.queue.amazonaws.com)
sqs: (ap-northeast-1, ap-northeast-1.queue.amazonaws.com)
sqs: (ap-northeast-2, ap-northeast-2.queue.amazonaws.com)
sqs: (ap-northeast-3, ap-northeast-3.queue.amazonaws.com)
sqs: (ap-south-1, ap-south-1.queue.amazonaws.com)
sqs: (ap-southeast-1, ap-southeast-1.queue.amazonaws.com)
sqs: (ap-southeast-2, ap-southeast-2.queue.amazonaws.com)
sqs: (ap-southeast-3, ap-southeast-3.queue.amazonaws.com)
sqs: (ca-central-1, ca-central-1.queue.amazonaws.com)
sqs: (eu-central-1, eu-central-1.queue.amazonaws.com)
sqs: (eu-north-1, eu-north-1.queue.amazonaws.com)
sqs: (eu-south-1, eu-south-1.queue.amazonaws.com)
sqs: (eu-west-1, eu-west-1.queue.amazonaws.com)
sqs: (eu-west-2, eu-west-2.queue.amazonaws.com)
sqs: (eu-west-3, eu-west-3.queue.amazonaws.com)
sqs: (me-south-1, me-south-1.queue.amazonaws.com)
sqs: (sa-east-1, sa-east-1.queue.amazonaws.com)
sqs: (us-east-1, queue.amazonaws.com)
sqs: (us-east-2, us-east-2.queue.amazonaws.com)
sqs: (us-west-1, us-west-1.queue.amazonaws.com)
sqs: (us-west-2, us-west-2.queue.amazonaws.com)
sqs: (cn-north-1, cn-north-1.queue.amazonaws.com.cn)
sqs: (cn-northwest-1, cn-northwest-1.queue.amazonaws.com.cn)
sqs: (us-gov-west-1, us-gov-west-1.queue.amazonaws.com)
sqs: (us-isob-east-1, us-isob-east-1.queue.sc2s.sgov.gov)

Reply:
Yeah, we definitely need to make sure the list is comprehensive if we're going with this approach. Can we be explicit and just generate all possible combinations for the 3 services. That can either live here or in a separate lookup json file.

i.e.
```json
{
   "elasticmapreduce": {
     "af-south-1": "af-south-1.elasticmapreduce.amazonaws.com",
     ...
   },{
   "sqs": {
     ...
   },
   ...
}
```

We also need to consider that not all regions are in a services endpoints.json entry, so let's make sure the region list is complete.

Reply:
just pushed up some updates. every service/region combination listed above is now included and hardcoded as a dictionary in the file